packages/repository/src/repository-builder.ts

```diff
@@ -0,0 +1,163 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/rest-crud
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import assert from 'assert';
+import {Constructor} from './common-types';
+import {Entity, Model} from './model';
+import {
+  CrudRepository,
+  DefaultCrudRepository,
+  DefaultKeyValueRepository,
+  EntityCrudRepository,
+  juggler,
+  KeyValueRepository,
+} from './repositories';
+
+/**
+ * Create (define) an entity CRUD repository class for the given model.
+ *
+ * @example
+ *
+ * ```ts
+ * const ProductRepository = defineEntityCrudRepositoryClass(Product);
+ * ```
+ *
+ * @param entityClass - An entity class such as `Product`.
+ * @param baseRepositoryClass - Base repository class. Defaults to `DefaultCrudRepository`
+ *
+ * @typeParam E - An entity class
+ * @typeParam IdType - ID type for the entity
+ * @typeParam Relations - Relations for the entity
+ */
+export function defineEntityCrudRepositoryClass<
+  T extends Entity,
```

Comment:
Can you change `T` to `E` for consistency with the docs and the interface below?